service/history/shardController.go

```diff
@@ -42,24 +41,19 @@ const (
 
 type (
 	shardController struct {
-		service             service.Service
-		host                *membership.HostInfo
-		hServiceResolver    membership.ServiceResolver
-		membershipUpdateCh  chan *membership.ChangedEvent
-		shardMgr            persistence.ShardManager
-		historyV2Mgr        persistence.HistoryManager
-		executionMgrFactory persistence.ExecutionManagerFactory
-		domainCache         cache.DomainCache
-		engineFactory       EngineFactory
-		shardClosedCh       chan int
-		isStarted           int32
-		isStopped           int32
-		shutdownWG          sync.WaitGroup
-		shutdownCh          chan struct{}
-		logger              log.Logger
-		throttledLoggger    log.Logger
-		config              *Config
-		metricsClient       metrics.Client
+		resource.Resource
+
+		membershipUpdateCh chan *membership.ChangedEvent
+		engineFactory      EngineFactory
+		shardClosedCh      chan int
+		isStarted          int32
+		isStopped          int32
```

Comment:
plz use the daemon status, commonly used in entities which have `Start` and `Stop` functions

Reply:
Done. Also modified `DescribeHistoryHost` a little bit.